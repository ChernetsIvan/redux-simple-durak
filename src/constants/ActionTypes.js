export const START_GAME_CLICKED = 'START_GAME_CLICKED';
export const BEGIN_GAME_CLICKED = 'BEGIN_GAME_CLICKED';
