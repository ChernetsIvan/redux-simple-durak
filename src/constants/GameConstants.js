export const cardRanks = [
  { text: "6", cardValue: 1 },
  { text: "7", cardValue: 2 },
  { text: "8", cardValue: 3 },
  { text: "9", cardValue: 4 },
  { text: "10", cardValue: 5 },
  { text: "В", cardValue: 6 },
  { text: "Д", cardValue: 7 },
  { text: "К", cardValue: 8 },
  { text: "Т", cardValue: 9 }
];

export const cardSuits = [
  { suit: "Ч" },
  { suit: "К" },
  { suit: "П" },
  { suit: "Б" }
];

export const statusMessages = {
  PlayerAttack: "Ваш ход!",
  PlayerDefence: "Защищайтесь!",
  PlayerDiscard: "Беру! Подбрасывать будете?",
  PlayerLose: "Вы проиграли! Сыграем ещё раз ? :)",
  PlayerWin: "Вы выиграли! Поздравляю :)",
  GameResultDraw: "Ничья!"
};
