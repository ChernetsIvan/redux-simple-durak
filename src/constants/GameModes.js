export const PlayerAttack = "PlayerAttack";
export const PlayerDefence = "PlayerDefence";
export const PlayerDiscard = "PlayerDiscard"; //AI забирает: игрок подбрасывает ему карты
export const AiAttack = "AiAttack";
export const AiDefence = "AiDefence";
export const AiDiscard = "AiDiscard"; //Игрок забирает: AI подбрасывает ему карты

export const PlayerWin = "PlayerWin";
export const AIWin = "AIWin";
export const TheDraw = "TheDraw"; //Ничья