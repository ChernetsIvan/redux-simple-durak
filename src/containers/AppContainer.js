import { connect } from 'react-redux';

import App from './../components/App';

const mapStateToProps = store => ({
    isRenderSettingsForStartNewGame: store.game.isRenderSettingsForStartNewGame
});

export default connect(mapStateToProps)(App);