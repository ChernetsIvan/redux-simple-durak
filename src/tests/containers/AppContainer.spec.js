import React from "react";
import renderer from "react-test-renderer";

import { Provider } from "react-redux";
import { createStore } from "redux";

import AppContainer from "./../../containers/AppContainer";
import reducer from "./../../reducers";
import * as actionTypes from "./../../constants/ActionTypes";

describe("AppContainer snapshot testing", () => {
  let store;

  beforeEach(() => {
    store = createStore(reducer);
  });

  test("Store in initial-state mode", () => {
    const component = renderer.create(
      <Provider store={store}>
        <AppContainer />
      </Provider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  test("Dispatch with START_GAME_CLICKED-action", () => {
    store.dispatch({
      type: actionTypes.START_GAME_CLICKED
    });
    const component = renderer.create(
      <Provider store={store}>
        <AppContainer />
      </Provider>
    );
    expect(component.toJSON().props).toMatchSnapshot();
  });

  test("Dispatch with BEGIN_GAME_CLICKED-action", () => {
    store.dispatch({
      type: actionTypes.BEGIN_GAME_CLICKED
    });
    const component = renderer.create(
      <Provider store={store}>
        <AppContainer />
      </Provider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  test("Dispatch with WHO_MOVE_FIRST_CHANGED-action, AI move first", () => {
    store.dispatch({
      type: actionTypes.WHO_MOVE_FIRST_CHANGED,
      payload: false
    });
    const component = renderer.create(
      <Provider store={store}>
        <AppContainer />
      </Provider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
