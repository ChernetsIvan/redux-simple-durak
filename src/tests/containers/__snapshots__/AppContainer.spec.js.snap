// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppContainer snapshot testing Dispatch with BEGIN_GAME_CLICKED-action 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <div
      className="row justify-content-center"
    >
      <div
        className="col-auto"
      >
        <h1
          className="display-2"
        >
          Карточная игра
        </h1>
      </div>
    </div>
    <div
      className="row justify-content-center mt-3"
    >
      <div
        className="col-auto"
      >
        <h1
          className="display-3"
        >
          Простой дурак
        </h1>
      </div>
    </div>
    <div
      className="row justify-content-center mt-5"
    >
      <div
        className="col-auto"
      >
        <span
          className="mr-4"
        >
          <b>
            Первым ходит:
          </b>
        </span>
        <span
          className="mr-2"
        >
          <input
            checked={true}
            className=""
            name="PlayerFirst"
            onChange={[Function]}
            type="radio"
          />
          <span
            className="ml-2"
          >
            Игрок
          </span>
        </span>
        <span
          className="mr-2"
        >
          <input
            checked={false}
            name="AiFirst"
            onChange={[Function]}
            type="radio"
          />
          <span
            className="ml-2"
          >
            Компьютер
          </span>
        </span>
      </div>
    </div>
    <div
      className="row justify-content-center mt-3"
    >
      <div
        className="col-auto"
      >
        <button
          className="btn btn-secondary mr-3"
          onClick={[Function]}
        >
          <span>
            —
          </span>
           Поехали!
        </button>
      </div>
    </div>
    <div
      className="row justify-content-center mt-3"
    >
      <div
        className="col-auto"
      >
        <a
          onClick={[Function]}
          style={Object {}}
        >
          Статистика игр
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`AppContainer snapshot testing Dispatch with START_GAME_CLICKED-action 1`] = `
<div>
  <div
    className="container  my-1"
  >
    <div
      className="row justify-content-between"
    >
      <div
        className="col-auto"
      >
        <button
          className="btn btn-secondary mr-3"
          onClick={[Function]}
        >
          Новая игра
        </button>
      </div>
      <div
        className="col-auto"
      >
        <h3
          className="mt-2"
        >
          <span>
            —
          </span>
           
          Ваш ход!
        </h3>
      </div>
      <div
        className="col-2"
      />
    </div>
  </div>
  <div>
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-1"
        />
        <div
          className="col-10"
        >
          <div
            className="container mt-1"
          >
            <div
              className="row justify-content-center"
            >
              <div
                className="col-1 ml-3 card mb-1 shirt"
              />
              <div
                className="col-1 ml-3 card mb-1 shirt"
              />
              <div
                className="col-1 ml-3 card mb-1 shirt"
              />
              <div
                className="col-1 ml-3 card mb-1 shirt"
              />
              <div
                className="col-1 ml-3 card mb-1 shirt"
              />
              <div
                className="col-1 ml-3 card mb-1 shirt"
              />
              <div
                className="col-1 align-self-center"
              />
            </div>
          </div>
        </div>
        <div
          className="col-1"
        />
      </div>
    </div>
    <div
      className="container"
    >
      <div
        className="row align-items-center"
      >
        <div
          className="col-10"
        >
          <div
            className="container invisible"
          >
            <div
              className="row"
            >
              <div
                className="col-1 card mb-1 shirt"
              />
            </div>
          </div>
          <div
            className="container invisible"
          >
            <div
              className="row"
            >
              <div
                className="col-1 card mb-1 shirt"
              />
            </div>
          </div>
        </div>
        <div
          className="col-2"
        >
          <div
            className="container"
          >
            <div
              className="row"
            >
              <div
                className="col-6 card mb-1"
                onClick={[Function]}
              >
                <div
                  className="row"
                >
                  <b
                    className="ml-1"
                  >
                    <b
                      className="blackText"
                    >
                      6
                    </b>
                  </b>
                </div>
                <div
                  className="row"
                >
                  <img
                    alt=""
                    className="littleSuitImg ml-1"
                    src="club.png"
                  />
                </div>
                <div
                  className="row justify-content-center"
                >
                  <img
                    alt=""
                    className="bigSuitImg my-2"
                    src="club.png"
                  />
                </div>
                <div
                  className="row justify-content-end"
                >
                  <img
                    alt=""
                    className="littleSuitImg rotated mr-1"
                    src="club.png"
                  />
                </div>
                <div
                  className="row justify-content-end"
                >
                  <b
                    className="rotated mr-1"
                  >
                    <b
                      className="blackText"
                    >
                      6
                    </b>
                  </b>
                </div>
              </div>
              <div
                className="col-6 card mb-1 shirt"
              />
            </div>
            <div
              className="row"
            >
              <div
                className="col-6"
              />
              <div
                className="col-6"
              >
                <div
                  className="container"
                >
                  <div
                    className="row justify-content-center"
                  >
                    <div
                      className="col-auto"
                    >
                      <b>
                        23
                      </b>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-1"
        />
        <div
          className="col-10"
        >
          <div
            className="container"
          >
            <div
              className="row justify-content-center"
            >
              <div
                className="col-1 align-self-center"
              />
              <div
                className="col-1 card mb-1"
                onClick={[Function]}
              >
                <div
                  className="row"
                >
                  <b
                    className="ml-1"
                  >
                    <b
                      className="blackText"
                    >
                      6
                    </b>
                  </b>
                </div>
                <div
                  className="row"
                >
                  <img
                    alt=""
                    className="littleSuitImg ml-1"
                    src="spade.png"
                  />
                </div>
                <div
                  className="row justify-content-center"
                >
                  <img
                    alt=""
                    className="bigSuitImg my-2"
                    src="spade.png"
                  />
                </div>
                <div
                  className="row justify-content-end"
                >
                  <img
                    alt=""
                    className="littleSuitImg rotated mr-1"
                    src="spade.png"
                  />
                </div>
                <div
                  className="row justify-content-end"
                >
                  <b
                    className="rotated mr-1"
                  >
                    <b
                      className="blackText"
                    >
                      6
                    </b>
                  </b>
                </div>
              </div>
              <div
                className="col-1 card mb-1"
                onClick={[Function]}
              >
                <div
                  className="row"
                >
                  <b
                    className="ml-1"
                  >
                    <b
                      className="redText"
                    >
                      7
                    </b>
                  </b>
                </div>
                <div
                  className="row"
                >
                  <img
                    alt=""
                    className="littleSuitImg ml-1"
                    src="diamond.png"
                  />
                </div>
                <div
                  className="row justify-content-center"
                >
                  <img
                    alt=""
                    className="bigSuitImg my-2"
                    src="diamond.png"
                  />
                </div>
                <div
                  className="row justify-content-end"
                >
                  <img
                    alt=""
                    className="littleSuitImg rotated mr-1"
                    src="diamond.png"
                  />
                </div>
                <div
                  className="row justify-content-end"
                >
                  <b
                    className="rotated mr-1"
                  >
                    <b
                      className="redText"
                    >
                      7
                    </b>
                  </b>
                </div>
              </div>
              <div
                className="col-1 card mb-1"
                onClick={[Function]}
              >
                <div
                  className="row"
                >
                  <b
                    className="ml-1"
                  >
                    <b
                      className="redText"
                    >
                      Т
                    </b>
                  </b>
                </div>
                <div
                  className="row"
                >
                  <img
                    alt=""
                    className="littleSuitImg ml-1"
                    src="heart.png"
                  />
                </div>
                <div
                  className="row justify-content-center"
                >
                  <img
                    alt=""
                    className="bigSuitImg my-2"
                    src="heart.png"
                  />
                </div>
                <div
                  className="row justify-content-end"
                >
                  <img
                    alt=""
                    className="littleSuitImg rotated mr-1"
                    src="heart.png"
                  />
                </div>
                <div
                  className="row justify-content-end"
                >
                  <b
                    className="rotated mr-1"
                  >
                    <b
                      className="redText"
                    >
                      Т
                    </b>
                  </b>
                </div>
              </div>
              <div
                className="col-1 card mb-1"
                onClick={[Function]}
              >
                <div
                  className="row"
                >
                  <b
                    className="ml-1"
                  >
                    <b
                      className="blackText"
                    >
                      9
                    </b>
                  </b>
                </div>
                <div
                  className="row"
                >
                  <img
                    alt=""
                    className="littleSuitImg ml-1"
                    src="club.png"
                  />
                </div>
                <div
                  className="row justify-content-center"
                >
                  <img
                    alt=""
                    className="bigSuitImg my-2"
                    src="club.png"
                  />
                </div>
                <div
                  className="row justify-content-end"
                >
                  <img
                    alt=""
                    className="littleSuitImg rotated mr-1"
                    src="club.png"
                  />
                </div>
                <div
                  className="row justify-content-end"
                >
                  <b
                    className="rotated mr-1"
                  >
                    <b
                      className="blackText"
                    >
                      9
                    </b>
                  </b>
                </div>
              </div>
              <div
                className="col-1 card mb-1"
                onClick={[Function]}
              >
                <div
                  className="row"
                >
                  <b
                    className="ml-1"
                  >
                    <b
                      className="blackText"
                    >
                      10
                    </b>
                  </b>
                </div>
                <div
                  className="row"
                >
                  <img
                    alt=""
                    className="littleSuitImg ml-1"
                    src="club.png"
                  />
                </div>
                <div
                  className="row justify-content-center"
                >
                  <img
                    alt=""
                    className="bigSuitImg my-2"
                    src="club.png"
                  />
                </div>
                <div
                  className="row justify-content-end"
                >
                  <img
                    alt=""
                    className="littleSuitImg rotated mr-1"
                    src="club.png"
                  />
                </div>
                <div
                  className="row justify-content-end"
                >
                  <b
                    className="rotated mr-1"
                  >
                    <b
                      className="blackText"
                    >
                      10
                    </b>
                  </b>
                </div>
              </div>
              <div
                className="col-1 card mb-1"
                onClick={[Function]}
              >
                <div
                  className="row"
                >
                  <b
                    className="ml-1"
                  >
                    <b
                      className="blackText"
                    >
                      К
                    </b>
                  </b>
                </div>
                <div
                  className="row"
                >
                  <img
                    alt=""
                    className="littleSuitImg ml-1"
                    src="club.png"
                  />
                </div>
                <div
                  className="row justify-content-center"
                >
                  <img
                    alt=""
                    className="bigSuitImg my-2"
                    src="club.png"
                  />
                </div>
                <div
                  className="row justify-content-end"
                >
                  <img
                    alt=""
                    className="littleSuitImg rotated mr-1"
                    src="club.png"
                  />
                </div>
                <div
                  className="row justify-content-end"
                >
                  <b
                    className="rotated mr-1"
                  >
                    <b
                      className="blackText"
                    >
                      К
                    </b>
                  </b>
                </div>
              </div>
              <div
                className="col-1 align-self-center"
              />
            </div>
          </div>
        </div>
        <div
          className="col-1"
        />
      </div>
      <div
        className="row justify-content-center"
      >
        <div
          className="col-auto"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`AppContainer snapshot testing Dispatch with WHO_MOVE_FIRST_CHANGED-action 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <div
      className="row justify-content-center"
    >
      <div
        className="col-auto"
      >
        <h1
          className="display-2"
        >
          Карточная игра
        </h1>
      </div>
    </div>
    <div
      className="row justify-content-center mt-3"
    >
      <div
        className="col-auto"
      >
        <h1
          className="display-3"
        >
          Простой дурак
        </h1>
      </div>
    </div>
    <div
      className="row justify-content-center mt-5"
    >
      <div
        className="col-auto"
      >
        <span
          className="mr-4"
        >
          <b>
            Первым ходит:
          </b>
        </span>
        <span
          className="mr-2"
        >
          <input
            checked={false}
            className=""
            name="PlayerFirst"
            onChange={[Function]}
            type="radio"
          />
          <span
            className="ml-2"
          >
            Игрок
          </span>
        </span>
        <span
          className="mr-2"
        >
          <input
            checked={true}
            name="AiFirst"
            onChange={[Function]}
            type="radio"
          />
          <span
            className="ml-2"
          >
            Компьютер
          </span>
        </span>
      </div>
    </div>
    <div
      className="row justify-content-center mt-3"
    >
      <div
        className="col-auto"
      >
        <button
          className="btn btn-secondary mr-3"
          onClick={[Function]}
        >
          <span>
            —
          </span>
           Поехали!
        </button>
      </div>
    </div>
    <div
      className="row justify-content-center mt-3"
    >
      <div
        className="col-auto"
      >
        <a
          onClick={[Function]}
          style={Object {}}
        >
          Статистика игр
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`AppContainer snapshot testing Dispatch with WHO_MOVE_FIRST_CHANGED-action, AI move first 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <div
      className="row justify-content-center"
    >
      <div
        className="col-auto"
      >
        <h1
          className="display-2"
        >
          Карточная игра
        </h1>
      </div>
    </div>
    <div
      className="row justify-content-center mt-3"
    >
      <div
        className="col-auto"
      >
        <h1
          className="display-3"
        >
          Простой дурак
        </h1>
      </div>
    </div>
    <div
      className="row justify-content-center mt-5"
    >
      <div
        className="col-auto"
      >
        <span
          className="mr-4"
        >
          <b>
            Первым ходит:
          </b>
        </span>
        <span
          className="mr-2"
        >
          <input
            checked={false}
            className=""
            name="PlayerFirst"
            onChange={[Function]}
            type="radio"
          />
          <span
            className="ml-2"
          >
            Игрок
          </span>
        </span>
        <span
          className="mr-2"
        >
          <input
            checked={true}
            name="AiFirst"
            onChange={[Function]}
            type="radio"
          />
          <span
            className="ml-2"
          >
            Компьютер
          </span>
        </span>
      </div>
    </div>
    <div
      className="row justify-content-center mt-3"
    >
      <div
        className="col-auto"
      >
        <button
          className="btn btn-secondary mr-3"
          onClick={[Function]}
        >
          <span>
            —
          </span>
           Поехали!
        </button>
      </div>
    </div>
    <div
      className="row justify-content-center mt-3"
    >
      <div
        className="col-auto"
      >
        <a
          onClick={[Function]}
          style={Object {}}
        >
          Статистика игр
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`AppContainer snapshot testing Dispatch with WHO_MOVE_FIRST_CHANGED-action, Player move first 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <div
      className="row justify-content-center"
    >
      <div
        className="col-auto"
      >
        <h1
          className="display-2"
        >
          Карточная игра
        </h1>
      </div>
    </div>
    <div
      className="row justify-content-center mt-3"
    >
      <div
        className="col-auto"
      >
        <h1
          className="display-3"
        >
          Простой дурак
        </h1>
      </div>
    </div>
    <div
      className="row justify-content-center mt-5"
    >
      <div
        className="col-auto"
      >
        <span
          className="mr-4"
        >
          <b>
            Первым ходит:
          </b>
        </span>
        <span
          className="mr-2"
        >
          <input
            checked={true}
            className=""
            name="PlayerFirst"
            onChange={[Function]}
            type="radio"
          />
          <span
            className="ml-2"
          >
            Игрок
          </span>
        </span>
        <span
          className="mr-2"
        >
          <input
            checked={false}
            name="AiFirst"
            onChange={[Function]}
            type="radio"
          />
          <span
            className="ml-2"
          >
            Компьютер
          </span>
        </span>
      </div>
    </div>
    <div
      className="row justify-content-center mt-3"
    >
      <div
        className="col-auto"
      >
        <button
          className="btn btn-secondary mr-3"
          onClick={[Function]}
        >
          <span>
            —
          </span>
           Поехали!
        </button>
      </div>
    </div>
    <div
      className="row justify-content-center mt-3"
    >
      <div
        className="col-auto"
      >
        <a
          onClick={[Function]}
          style={Object {}}
        >
          Статистика игр
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`AppContainer snapshot testing Store in initial-state mode 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <div
      className="row justify-content-center"
    >
      <div
        className="col-auto"
      >
        <h1
          className="display-2"
        >
          Карточная игра
        </h1>
      </div>
    </div>
    <div
      className="row justify-content-center mt-3"
    >
      <div
        className="col-auto"
      >
        <h1
          className="display-3"
        >
          Простой дурак
        </h1>
      </div>
    </div>
    <div
      className="row justify-content-center mt-5"
    >
      <div
        className="col-auto"
      >
        <span
          className="mr-4"
        >
          <b>
            Первым ходит:
          </b>
        </span>
        <span
          className="mr-2"
        >
          <input
            checked={true}
            className=""
            name="PlayerFirst"
            onChange={[Function]}
            type="radio"
          />
          <span
            className="ml-2"
          >
            Игрок
          </span>
        </span>
        <span
          className="mr-2"
        >
          <input
            checked={false}
            name="AiFirst"
            onChange={[Function]}
            type="radio"
          />
          <span
            className="ml-2"
          >
            Компьютер
          </span>
        </span>
      </div>
    </div>
    <div
      className="row justify-content-center mt-3"
    >
      <div
        className="col-auto"
      >
        <button
          className="btn btn-secondary mr-3"
          onClick={[Function]}
        >
          <span>
            —
          </span>
           Поехали!
        </button>
      </div>
    </div>
    <div
      className="row justify-content-center mt-3"
    >
      <div
        className="col-auto"
      >
        <a
          onClick={[Function]}
          style={Object {}}
        >
          Статистика игр
        </a>
      </div>
    </div>
  </div>
</div>
`;
